src/main/java/ua/com/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -50,4 +51,13 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataProcessingException("Can't get all cinema halls", e);
         }
     }
+
+    @Override
+    public Optional<CinemaHall> getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.load(CinemaHall.class, id));
```

Comment:
please look carefully at the previous comment about get()/load()
it was about delete() method

in get() methods - is it reasonable to wrap the proxy in Optional?

Reply:
my bad😢